code/modules/wiremod/components/list/format.dm

```diff
@@ -111,3 +79,39 @@
 
 /obj/item/circuit_component/format/assoc/make_params_port()
 	param_list_port = add_input_port("Params", PORT_TYPE_ASSOC_LIST(PORT_TYPE_STRING, PORT_TYPE_ANY))
+
+/**
+ * # Subtype of regex that holds context to /obj/item/circuit_component/format
+ */
+/regex/format_component
+	var/obj/item/circuit_component/format/context
```

Comment:
what happens when the circuit component is deleted? and it still holds a reference? It might lead to a harddel, you may want to unassign the context after the proc has executed

Reply:
ah, good point!

Reply:
done